### DISCUSSION TITLE
Control Transmission Path for TSN through ini File

### DISCUSSION THREAD

**Discussion by m5awi** [inet-framework/inet#969] [General]
Hello everyone,

Is there a way to set the transmission path, through which the stream is used in TSN, in an ini file? I am seeking a method similar to the one presented in the streamRedundancyConfigurator, such as [''n1", "s1", "n2"].  Currently, I am encountering the issue that the stream is automatically chooses the closest path to the destination, but I would like to have the ability to control this path.

My second question pertains to the streamRedundancyConfigurator. Whenever there is one stream that must flow through disjoint paths, it doesn't work as expected. Duplication of packets only functions if the two redundancy paths use the same switch, but it does not work if the paths are disjoint. For instance: [[''n1", "s1", "n2"],["n1","s2","n2"]].

Thank you.

**Comment by levy** [comment]
You can fully control the path from the INI file by manually configuring the StreamSplitter, StreamMerger, StreamEncoding, StreamDecoding, MacForwardingTable, etc. modules in all nodes, or use the StreamRedundancyConfigurator, which does that for you even if the path is not redundant.

You should create an issue to report a bug, preferably with a simple example to reproduce it.
